src/sentry/ingest/transaction_clusterer/tasks.py

```diff
@@ -47,28 +47,36 @@ def spawn_clusterers(**kwargs: Any) -> None:
     queue="transactions.name_clusterer",
     default_retry_delay=5,  # copied from release monitor
     max_retries=5,  # copied from release monitor
+    soft_time_limit=5,
+    time_limit=7,  # extra 2s to emit metrics
```

Comment:
Tasks take ~2.5s on average, with max peaks of ~3.5s. A 5s limit is rather conservative to not trigger any thresholds in the current state.

Reply:
1. Most other Celery tasks have a much larger timeout, I would feel more comfortable if we took ~4x the average, i.e. 10 seconds.
2. Can we make these timeouts depend on `PROJECTS_PER_TASK`? E.g.
```suggestion
    soft_time_limit=0.1 * PROJECTS_PER_TASK, # 100 ms per project
    time_limit=0.1 * PROJECTS_PER_TASK + 2,  # extra 2s to emit metrics
```